File: Practica1/Examen/QytetetRubyPruebaExamen/pkg/QytetetRuby-0.0.1/lib/titulo_propiedad.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'jugador'
require_relative 'casilla'

module ModeloQytetet
  class TituloPropiedad
    #Modificadores y consultores
    attr_accessor :hipotecada
    attr_reader :nombre, :alquiler_base, :factor_revalorizacion, :hipoteca_base, :precio_edificar
    attr_writer :casilla, :propietario
    
    def initialize(n, aBase, fReva, hBase, pEdi)
      @nombre = n
      @alquiler_base = aBase
      @factor_revalorizacion = fReva
      @hipoteca_base = hBase
      @precio_edificar = pEdi
      @hipotecada = false
      @propietario = nil
      @casilla = nil
    end
    
    #Métodos
    def cobrar_alquiler(coste)
      
    end
    
    def propietario_encarcelado()
      
    end
    
    def tengo_propietario()
      
    end
    
    #Método toString() devuelve un String con el estado del objeto correspondiente
    def to_s
        "\n Nombre: #{@nombre}
        \n Esta hipotecada?: #{@hipotecada}
        \n Precio base alquiler: #{@alquiler_base} 
        \n Factor de revalorizacion: #{@factor_revalorizacion}
        \n Valor base hipoteca: #{@hipoteca_base}
        \n Precio de edificacion: #{@precio_edificar} \n"
    end
  end
end

File: Practica1/QytetetJava/src/modeloqytetet/TituloPropiedad.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author Lorena
 */
public class TituloPropiedad {
    private String nombre; //nombre de la calle
    private boolean hipotecada; //indica si esta hipotecada la propiedad o no
    private int alquilerBase; //precio base (sin tener en cuenta las edificaciones) que debe pagar quien caiga en la casilla.
    private float factorRevalorizacion; //indica cuánto se revaloriza el título de propiedad en el periodo transcurrido entre su compra y su venta.
    private int hipotecaBase; //indica cuál es el valor base de su hipoteca,
    private int precioEdificar; //indica cuánto cuesta edificar casas y hoteles,

    //Constructor
    public TituloPropiedad(String nombre, int alquilerBase, float factorRevalorizacion, int hipotecaBase, int precioEdificar){
        this.nombre = nombre;
        hipotecada = false;
        this.alquilerBase = alquilerBase;
        this.factorRevalorizacion = factorRevalorizacion;
        this.hipotecaBase = hipotecaBase;
        this.precioEdificar = precioEdificar;
    }
    
    //Modificador 
    void setHipotecada(boolean condicion){
        hipotecada = condicion;
    }
    
    //Consultores
    boolean getHipotecada(){
        return hipotecada;
    }
    
    String getNombre(){
        return nombre;
    }
    
    int getAlquilerBase(){
        return alquilerBase;
    }
    
    float getFactorRevalorizacion(){
        return factorRevalorizacion;
    }
    
    int getHipotecaBase(){
        return hipotecaBase;
    }
    
    int getPrecioEdificar(){
        return precioEdificar;
    }
    
    //Método toString() devuelve un String con el estado del objeto correspondiente
    @Override
    public String toString(){
        return " Nombre:" + nombre 
                + "\n Esta hipotecada?:" + Boolean.toString(hipotecada)
                + "\n Precio base alquiler:" + Integer.toString(alquilerBase) 
                + "\n Factor de revalorizacion:" + Float.toString(factorRevalorizacion)
                + "\n Valor base hipoteca:" + Integer.toString(hipotecaBase) 
                + "\n Precio de edificacion:" + Integer.toString(precioEdificar);
    }
    
}    
